src/org/zaproxy/zap/extension/websocket/pscan/scripts/ScriptsWebSocketPassiveScanner.java

```diff
@@ -0,0 +1,192 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.websocket.pscan.scripts;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.zaproxy.zap.extension.script.ExtensionScript;
+import org.zaproxy.zap.extension.script.ScriptType;
+import org.zaproxy.zap.extension.script.ScriptWrapper;
+import org.zaproxy.zap.extension.websocket.ExtensionWebSocket;
+import org.zaproxy.zap.extension.websocket.WebSocketMessageDTO;
+import org.zaproxy.zap.extension.websocket.alerts.WebSocketAlertWrapper;
+import org.zaproxy.zap.extension.websocket.alerts.WebSocketPassiveScanAlert;
+import org.zaproxy.zap.extension.websocket.pscan.WebSocketPassiveScanThread;
+import org.zaproxy.zap.extension.websocket.pscan.WebSocketPassiveScannerManager;
+import org.zaproxy.zap.extension.websocket.pscan.WebSocketPluginPassiveScanner;
+
+import java.lang.reflect.UndeclaredThrowableException;
+import java.util.List;
+
+/**
+ * Implement a plugin for Passive Scanning Scripting.
+ */
+public class ScriptsWebSocketPassiveScanner extends WebSocketPluginPassiveScanner {
+    
+    private static final Logger LOGGER = Logger.getLogger(ScriptsWebSocketPassiveScanner.class);
+    public static final String PLUGIN_NAME = "WS.ScriptPassiveScan";
+    public static final int PLUGIN_ID = 25379412;
+    
+    private boolean isEnabled;
+    private ExtensionScript extensionScript;
+    
+    /**
+     * Implement a plugin for Passive Scanning Scripting.
+     * Should have been Registered the {@link ScriptType} at {@link ExtensionScript}.
+     * By default the plug in is disabled. To enable the plugin
+     * use the {@link WebSocketPassiveScannerManager} or in any
+     * other case use {@link ScriptsWebSocketPassiveScanner#setEnabled(boolean)}
+     */
+    public ScriptsWebSocketPassiveScanner(){
+        isEnabled = false;
+    }
+    
+    private ExtensionScript getExtension() {
+        if (extensionScript == null) {
+            extensionScript = Control.getSingleton().getExtensionLoader().getExtension(ExtensionScript.class);
+        }
+        return extensionScript;
+    }
+    
+    private WebSocketPassiveScanAlert getScriptAlert(WebSocketPassiveScanThread thread){
+        return new AlertRaiser(thread);
+    }
+    
+    @Override
+    public void scanMessage(WebSocketPassiveScanThread thread, WebSocketMessageDTO webSocketMessage) {
+        if(getExtension() != null){
+            List<ScriptWrapper> scriptWrappers = extensionScript.getScripts(ExtensionWebSocket.SCRIPT_TYPE_WEBSOCKET_PASSIVE);
+            WebSocketPassiveScanAlert alertRaiser = getScriptAlert(thread);
```

Comment:
I am cautious about that. Maybe create every time a new Object it is not a good idea